src/pages/blog/partial-hydration.mdx

```diff
@@ -0,0 +1,238 @@
+
+So how can NextJS give an application the ability to behave like it is both rendered on the server _and_ on the client? Long story short: it truly just does both. When you request a page from NextJS, Next fetches the initial data that will be required to render that page, and then constructs the first render for that page as if it was running on the client. Then Next, under the hood, passes that React component tree to a function called `{ renderToString } from "react-dom/server"` (✳️ 2). It then places that HTML string in a greater HTML body and then injects the React application as a script tag as well as an _extra_ script tag labeled `<script id="__NEXT_DATA__" type="application/json">` which contains the async fetched data used to get the first render (this is the data that is returned from `getServerSideProps` or `getStaticProps`).
+
+When the client receives your page, because it has prerendered HTML already available, it is able to render that HTML almost immediately. This means that users requesting a NextJS site are likely going to see and start mentally interpretting content earlier than a React SPA. However, for a brief period of time this page is still not interactive. Any logic that required JavaScript to be run based on user actions (e.g. event handlers) will not work at all. When loading on the client, a function called `{ hydrateRoot } from "react-dom/client"` mounts your React application to the same DOM root that it was placed on when the server rendered it. Once your React app has been "hydrated" with it's client side logic, it starts behaving exactly like an SPA.
+
+## The problems with the full hydration of the NextJS pages directory
+
+The period of non-interactivity (pre-hydration) can potentially leave a user with a page that looks like it works, but doesn't respond to user interaction for a period of time. The user could theoretically feel like this is a performance issue with the site or application. However, This period of time is generally considered to be non-problematic and _usually_ to a user's benefit. A user with a quick internet connection can get a readable version of the page quicker and start using the site while the hydration process is happening in the background. Most users aren't going to be needing interactivity in the first second or so as they start navigating a web page as they must first _mentally_ index it.
```

Comment:
I might be getting ahead of myself here, but is it possible to give a concise explanation of what "hydration" is so that the reader is on the same page?

I understand it as a magical process whereby static HTML is converted into React's virtual DOM and appropriate event handlers are all hooked up.

Reply:
Added a deeper (but still lightweight) desc to the previous paragraph